packages/pigeon/lib/swift_generator.dart

```diff
@@ -607,6 +589,75 @@ import FlutterMacOS
     indent.newln();
   }
 
+  /// Writes decode and casting code for any type.
+  ///
+  /// Optional parameters are necessary for class decoding only.
+  void _writeDecodeCasting({
+    required Root root,
+    required Indent indent,
+    required String value,
+    required String variableName,
+    required TypeDeclaration type,
+    Set<String>? customClassNames,
+    Set<String>? customEnumNames,
+  }) {
+    String castForceUnwrap(String value, TypeDeclaration type, Root root) {
+      if (isEnum(root, type)) {
+        assert(!type.isNullable,
+            'nullable enums require special code that this helper does not supply');
+        return '${_swiftTypeForDartType(type)}(rawValue: $value as! Int)!';
+      } else if (type.baseName == 'Object') {
+        // Special-cased to avoid warnings about using 'as' with Any.
+        return value;
+      } else if (type.baseName == 'int') {
+        final String orString =
+            type.isNullable ? 'nilOrValue($value)' : '$value as! Int64';
+        return '($value is Int32) ? Int64($value as! Int32) : $orString';
```

Comment:
It occurs to me that we should reverse the conditional based on what I found about `NSNumber`; as long as the underlying value actually is an `NSNumber`, "casting" to `Int64` directly will work, so checking for `Int32` first (which will also work) will result in two conversions (`NSNumber` -> `Int32` > `Int64`) instead of one.

So:
```
final String int64String = type.isNullable ? 'nilOrValue($value)' : '$value as! Int64';
return '($value is Int64) ? $int64String : Int64($value as! Int32);
```

Reply:
That doesn't work for nullables though

Reply:
And I'm fairly certain it doesn't lower the conversion amount, since casting checking the cast type doesn't perform an actual casting

Reply:
are you sure this line `let foo : Int64 = $value is Int64 ? $value as! Int64 : Int64($value as! Int32)` doesn't also convert to Int64 two times? It's a little hard to look up how this works